src/core/openthread-core-config.h

```diff
@@ -34,14 +34,10 @@
 #ifndef OPENTHREAD_CORE_CONFIG_H_
 #define OPENTHREAD_CORE_CONFIG_H_
 
-#ifndef OPENTHREAD_CORE_HAVE_PROJECT_SPECIFIC_CONFIG
-#define OPENTHREAD_CORE_HAVE_PROJECT_SPECIFIC_CONFIG 0
-#endif
-
 #define OPENTHREAD_CORE_CONFIG_H_IN
 
-#if OPENTHREAD_CORE_HAVE_PROJECT_SPECIFIC_CONFIG
-#include <openthread-core-project-config.h>
+#ifdef OPENTHREAD_PORJECT_CORE_CONFIG_FILE
```

Comment:
But, why would this change break you? Couldn't you define `OPENTHREAD_CORE_HAVE_PROJECT_SPECIFIC_CONFIG` as `"openthread-core-project-config.h"` and it should be the same thing for you?